python-avd/pyavd/_eos_designs/structured_config/network_services/vxlan_interface.py

```diff
@@ -24,152 +26,91 @@ class VxlanInterfaceMixin(Protocol):
-        vnis: list[dict] = []
+            vxlan.controller_client.enabled = True
 
         for tenant in self.shared_utils.filtered_tenants:
             for vrf in tenant.vrfs:
-                self._get_vxlan_interface_config_for_vrf(vrf, tenant, vrfs, vlans, vnis)
+                self._update_vxlan_interface_config_for_vrf(vrf, tenant)
 
             if not self.shared_utils.network_services_l2:
                 continue
 
             for l2vlan in tenant.l2vlans:
-                if vlan := self._get_vxlan_interface_config_for_vlan(l2vlan, tenant):
-                    # Duplicate check is not done on the actual list of vlans, but instead on our local "vnis" list.
-                    # This is necessary to find duplicate VNIs across multiple object types.
-                    append_if_not_duplicate(
-                        list_of_dicts=vnis,
-                        primary_key="vni",
-                        new_dict=vlan,
-                        context="VXLAN VNIs for L2VLANs",
-                        context_keys=["id", "name", "vni"],
-                    )
-                    # Here we append to the actual list of VRFs, so duplication check is on the VLAN ID here.
-                    append_if_not_duplicate(
-                        list_of_dicts=vlans,
-                        primary_key="id",
-                        new_dict=vlan,
-                        context="VXLAN VNIs for L2VLANs",
-                        context_keys=["id", "vni"],
-                    )
+                if l2vlan.vxlan:
+                    self._update_vxlan_interface_config_for_vlan(l2vlan, tenant)
 
         if self.shared_utils.is_wan_server:
             # loop through wan_vrfs and add VRF VNI if not present
             for vrf in self._filtered_wan_vrfs:
-                # Duplicate check is not done on the actual list of vlans, but instead on our local "vnis" list.
-                # This is necessary to find duplicate VNIs across multiple object types.
-                vrf_data = {"name": vrf.name, "vni": vrf.wan_vni}
-                append_if_not_duplicate(
-                    list_of_dicts=vnis,
-                    primary_key="vni",
-                    new_dict=vrf_data,
-                    context="VXLAN VNIs for VRFs",
-                    context_keys=["id", "name", "vni"],
-                )
-                # Here we append to the actual list of VRFs, so duplication check is on the VRF here.
-                append_if_not_duplicate(
-                    list_of_dicts=vrfs,
-                    primary_key="name",
-                    new_dict=vrf_data,
-                    context="VXLAN VNIs for VRFs",
-                    context_keys=["name", "vni"],
-                )
-
-        if vlans:
-            vxlan["vlans"] = vlans
+                vxlan.vrfs.append_new(name=vrf.name, vni=vrf.wan_vni)
```

Comment:
It is a bit confusing that we also add VRFs here - especially since it could be the same VRFs as added in the other logic. So we could easily end up with conflicts, since the other one takes more variables into account. Can we combine them? (Optional since it already works like this, but when we do gateway this will more likely break)

Reply:
I will close this comment, since the behavior is the same as Today. Approved the PR.